src/app.ts

```diff
@@ -47,6 +47,10 @@ export function generateServerlessRouter(fhirConfig: FhirConfig, supportedGeneri
                 accessToken,
             });
             if (isAllowed) {
+                res.locals.allowedResourceTypes = fhirConfig.auth.authorization.getAllowedResourceTypesForOperation({
+                    operation: requestInformation.operation,
+                    accessToken,
+                });

```

Comment:
Should we only call this if we need this (include/revinclude Search)? Perhaps we could use outgoing chaining? Like how we are doing the error handling (not sure if possible, to follow that express paradigm here)

Something like this middle ware: https://dev.to/getd/express-middleware-is-just-a-fancy-way-of-chaining-a-bunch-of-functions-explained-in-3-mins-43jf

Reply:
I refactored it so that is called exclusively by search requests.

I don't think that resource filtering as middleware is a great idea. I don't like the idea of having the router being concerned with filtering search results. Plus, it is much more efficient to send the allowedResourceTypes to search since that can optimize the query plan(i.e. wildcards no longer need to query ALL resources. only the ones that are allowed)